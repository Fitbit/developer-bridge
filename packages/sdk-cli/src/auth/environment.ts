import process from 'process';
// tslint:disable-next-line:import-name
import environments from './environments.json';

export default () => {
  const defaultEnv = 'production';
  const environment = (
    process.env.FITBIT_SDK_ENVIRONMENT || defaultEnv
  ) as keyof typeof environments;

  if (Object.keys(environments).indexOf(environment) === -1) {
    throw new Error(`Invalid environment specified: ${environment}`);
  }

  const config = environments[environment];

  const clientId = process.env.FITBIT_SDK_CLIENT_ID;
  const clientSecret = process.env.FITBIT_SDK_CLIENT_SECRET;

  if (clientId || clientSecret) {
    if (!clientId || !clientSecret) {
      throw new Error('Both FITBIT_SDK_CLIENT_ID and FITBIT_SDK_CLIENT_SECRET must be specified');
    }

    config.clientId = clientId;
    config.clientSecret = clientSecret;
  }

  return {
    environment,
    config,
  };
};
