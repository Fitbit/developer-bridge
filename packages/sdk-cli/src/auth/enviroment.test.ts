import environment from './environment';

function isValidEnv(env: string) {
  process.env.FITBIT_SDK_ENVIRONMENT = env;
  expect(environment()).toEqual(
    expect.objectContaining({
      environment: env,
    }),
  );
}

function isInvalidEnv(env: string) {
  process.env.FITBIT_SDK_ENVIRONMENT = env;
  expect(environment).toThrowErrorMatchingSnapshot();
}

beforeEach(() => {
  delete process.env.FITBIT_SDK_ENVIRONMENT;
  delete process.env.FITBIT_SDK_CLIENT_ID;
});

describe.each([
  'int',
  'stage',
  'production',
])('given an FITBIT_SDK_ENVIRONMENT value of "%s"', (env) => {
  test('returns environment data', () => isValidEnv(env));
});

describe.each([
  '__bad_env__',
])('given an FITBIT_SDK_ENVIRONMENT value of "%s"', (env) => {
  test('throws', () => isInvalidEnv(env));
});

it('defaults to the production environment', () => {
  expect(environment()).toEqual(
    expect.objectContaining({
      environment: 'production',
    }),
  );
});

it('uses FITBIT_SDK_CLIENT_ID if provided', () => {
  const clientId = '_fake_client_id_';

  process.env.FITBIT_SDK_CLIENT_ID = clientId;

  expect(environment()).toEqual(
    {
      config: expect.objectContaining({
        clientId,
      }),
      environment: 'production',
    },
  );
});
