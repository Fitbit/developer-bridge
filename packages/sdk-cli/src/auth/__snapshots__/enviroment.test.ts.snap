// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`given an FITBIT_SDK_ENVIRONMENT value of "__bad_env__" throws 1`] = `"Invalid environment specified: __bad_env__"`;

exports[`throws an error if only FITBIT_SDK_CLIENT_ID is set 1`] = `"Both FITBIT_SDK_CLIENT_ID and FITBIT_SDK_CLIENT_SECRET must be specified"`;

exports[`throws an error if only FITBIT_SDK_CLIENT_SECRET is set 1`] = `"Both FITBIT_SDK_CLIENT_ID and FITBIT_SDK_CLIENT_SECRET must be specified"`;
